### PR TITLE
Add sender to Sumo Logic exporter

### PR DESCRIPTION
**Pull request by sumo-drosiek**
Depends on #1

### CODE REVIEW

File: exporter/sumologicexporter/sender_test.go [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+	defer func() { test.srv.Close() }()
+	test.s.config.LogFormat = JSONFormat
+	test.s.config.MaxRequestBodySize = 10
+	test.s.buffer = exampleTwoLogs()
+
+	dropped, err := test.s.sendLogsJSONFormat("test_metadata")
+	assert.EqualError(t, err, "error during sending data: 500 Internal Server Error")
+	assert.Equal(t, test.s.buffer[0:1], dropped)
+}
+func TestSendJsonSplitFailedAll(t *testing.T) {
```
Comment (pmalek-sumo) [inline exporter/sumologicexporter/sender_test.go:250]:
```suggestion
}

func TestSendJsonSplitFailedAll(t *testing.T) {
```

File: exporter/sumologicexporter/sender_test.go [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+
+func TestBuffer(t *testing.T) {
+	test := prepareSenderTest(t, []func(w http.ResponseWriter, req *http.Request){})
+	defer func() { test.srv.Close() }()
+
+	assert.Equal(t, test.s.count(), 0)
+	logs := exampleTwoLogs()
+
+	droppedLogs, err := test.s.batch(logs[0], "")
+	require.Equal(t, err, nil)
```
Comment (pmalek-sumo) [inline exporter/sumologicexporter/sender_test.go:345]:
```suggestion
	require.NoError(t, err)
```

File: exporter/sumologicexporter/sender.go [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+	if _, err := body.WriteString(line); err != nil && appended {
+		errors = append(errors, err)
+		appended = false
+	}
+
+	if len(errors) > 0 {
+		return sent, appended, componenterror.CombineErrors(errors)
+	}
+	return sent, appended, nil
+}
```
Comment (pmalek-sumo) [inline exporter/sumologicexporter/sender.go:242]:
```suggestion
// appendAndSend appends line to the request body that will be sent and sends
// the accumulated data if the internal buffer has been filled (with maxBufferSize elements).
// It returns 2 booleans saying whether the data was sent and whether the provided data
// was appended and an error if anything failed.
func (s *sender) appendAndSend(
	line string,
	pipeline PipelineType,
	body *strings.Builder,
	fields Fields,
) (sent bool, appended bool, _ error) {
	var errors []error
	// appended keeps state of appending new log line to the body
	appended = true

	if body.Len() > 0 && body.Len()+len(line) > s.config.MaxRequestBodySize {
		sent = true
		if err := s.send(LogsPipeline, strings.NewReader(body.String()), fields); err != nil {
			errors = append(errors, err)
		}
		body.Reset()
	}

	if body.Len() > 0 {
		// Do not add newline if the body is empty
		if _, err := body.WriteString("\n"); err != nil {
			errors = append(errors, err)
			appended = false
		}
	}

	if _, err := body.WriteString(line); err != nil && appended {
		errors = append(errors, err)
		appended = false
	}

	if len(errors) > 0 {
		return sent, appended, componenterror.CombineErrors(errors)
	}
	return sent, appended, nil
}
```

Comment (sumo-drosiek) [reply exporter/sumologicexporter/sender.go:242]: I would rather keep sent as false explicit

Comment (sumo-drosiek) [reply exporter/sumologicexporter/sender.go]: Resolved on DM

**Pull request merged**